pytorch_lightning/strategies/ddp_spawn.py

```diff
@@ -373,7 +373,9 @@ def teardown(self) -> None:
         if isinstance(self.model, DistributedDataParallel):
             self.model = self.lightning_module
 
-        if self.sync_batchnorm:
+        trainer_fn = self.lightning_module.trainer.state.fn
```

Comment:
this is causing the test failures because the trainer instance is not available in teardown. @awaelchli do you know why that is? i thought `teardown` now runs inside of the spawned process as part of aligning the control flow

Reply:
For reference, the teardown was only added recently to the exception handling, in #11620
cc @carmocca 

An option to explore would be to set the stage in the top level method fit, validate, etc. Currently it is done in _fit_impl etc. which is running in the worker process.

Reply:
I agree. The loop teardowns also get missed because of this problem with ddp spawn

https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/trainer/trainer.py#L1250

Reply:
> I agree. 

I disagree. They are getting missed where? teardown runs fine in the worker processes, and in the main process where the trainer never trained, there is nothing to teardown for the loops.